treeherder/webapp/api/runnable_jobs.py

```diff
@@ -77,30 +77,29 @@ def list(self, request, project):
         for label, node in tc_graph.iteritems():
             build_platform = node['task']['extra']['treeherder']['build']['platform']
             job_type_name = node['task']['metadata']['name']
+            task_metadata = node['task']['metadata']
+            treeherder_options = node['task']['extra']['treeherder']
 
             # Not all tasks have a group name
-            if 'groupName' in node['task']['extra']['treeherder']:
-                job_group_name = node['task']['extra']['treeherder']['groupName']
-            else:
-                job_group_name = ""
+            job_group_name = treeherder_options.get('groupName', '')
+
             # Not all tasks have a group symbol
-            if 'groupSymbol' in node['task']['extra']['treeherder']:
-                job_group_symbol = node['task']['extra']['treeherder']['groupSymbol']
-            else:
-                job_group_symbol = ""
+            job_group_symbol = treeherder_options.get('groupSymbol', '')
+
             # Not all tasks have a collection
-            if 'collection' in node['task']['extra']['treeherder']:
-                platform_option = node['task']['extra']['treeherder']['collection'].keys()[0]
+            if 'collection' in treeherder_options:
+                platform_option = treeherder_options['collection'].keys()[0]
```

Comment:
I've just noticed that the buildbot implementation actually joins the options, if there are multiple, rather than just taking the first. If it was desired to have a consistent approach, this would do that:

```python
platform_option = ' '.join(treeherder_options['collection'].keys())
```

(Though interestingly, the `.keys()` isn't actually needed, since `.join()` appears to convert the dict to a list first, and `list(foo_dict)` only returns the keys - however it's a little less intuitive to read IMO)